Build a regression model.

In [5]:
# imports
import pandas as pd
import statsmodels.api as sm

grouped_business_df = pd.read_csv('grouped_business.csv', index_col = 0)

In [4]:
grouped_business_df

Station Name  Average Rating  Average Reviews  \
0                1 Ave & E 110 St           4.375           345.00   
1                 1 Ave & E 16 St           4.375          1545.35   
2                 1 Ave & E 18 St           4.400          1483.45   
3                 1 Ave & E 30 St           4.350          1066.85   
4                 1 Ave & E 62 St           4.300           713.90   
..                            ...             ...              ...   
995        Wolcott St & Dwight St           4.200           557.20   
996      Woodward Ave & Harman St           4.350           258.35   
997     Wyckoff Av & Jefferson St           4.300           311.30   
998      Wyckoff Av & Stanhope St           4.275           307.90   
999  Wythe Ave & Metropolitan Ave           4.175           827.80   

     Number of Bikes  
0                 24  
1                 51  
2                 49  
3                 35  
4                 31  
..               ...  
995               20  
996               16  
997               25  
998               21  
999               47  

[1000 rows x 4 columns]

In [9]:
X = grouped_business_df['Number of Bikes']
Y = grouped_business_df['Average Rating']

In [12]:
X = sm.add_constant(X)
lin_model = sm.OLS(y, X).fit()

In [15]:
model = lin_model.summary()

Provide model output and an interpretation of the results. 

In [17]:
print(model)

                            OLS Regression Results                            
Dep. Variable:         Average Rating   R-squared:                       0.032
Model:                            OLS   Adj. R-squared:                  0.031
Method:                 Least Squares   F-statistic:                     32.90
Date:                Wed, 30 Aug 2023   Prob (F-statistic):           1.29e-08
Time:                        13:37:52   Log-Likelihood:                 287.50
No. Observations:                1000   AIC:                            -571.0
Df Residuals:                     998   BIC:                            -561.2
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
const               4.1246      0.012    3

Interpretation of Results:

R-squared: From the model above, an R-squared value of 0.032 indicates that about 3.2% of the variability in the dependent variable (Average Rating) is explained by the variability in the independent variable (Number of Bikes). Thus, this model does not explain a significant amount og variation in the dependent variable.

P-value: In this model output, the P-values for both the const coefficient and the Number of Bikes coefficient are very small (close to zero). This indicates that they are both statistically significant predictors of the dependent variable, Average Rating.

Coefficient: For this linear regression model, the coefficient for Number of Bikes is approximately 0.0019, suggestitng that on average, a one unit increase in the Number of Bikes correlates with an increase of 0.0019 in the Average Rating.

Conclusion: While we can establish a statistically significant relationship between the independent and dependent variables, it is also worthy to note that the relationship between them is quite weak as evidenced by the low R-squared value.

# Stretch

How can you turn the regression model into a classification model?

To convert the regression model into a classification model, I will first define which attritube I am trying to predict. Next, I'll work on grouping mu continuous variables into discrete labels to help create the class. Then I'll select an appropriate classification algorithm and afterwards, evaluate my model.